File: add_animals.py
import json

new_lang = "tr"
kingdom = "animals"

with open(f"data/{kingdom}_{new_lang}.json", "r") as f:
    new_species = json.load(f)

with open(f"Species Lists/{kingdom}.json", "r") as f:
    old_species = json.load(f)

nodes = [species["nodeId"] for species in old_species]
new_nodes = [species["nodeId"] for species in new_species]

for new_node in new_nodes:
    if new_node not in nodes:
        species = new_species[new_nodes.index(new_node)]
        print(f"Node {new_node} not found in old species")
        print(species)


output = []

for species in old_species:
    node = species["nodeId"]
    if node in new_nodes:
        print(node)
        new_animal = new_species[new_nodes.index(node)]
        field = f"name_{new_lang}"
        new_name = new_animal[field]
        species[field] = new_name
        output.append(species)
    else:
        raise Exception(f"Node {node} not found in new animals")


with open(f"data/{kingdom}_added.json", "w") as f:
    json.dump(output, f, indent=4, ensure_ascii=False)

print("Done")

File: scripts/add_animals.ts
interface Species {
  nodeId: string;
  [key: string]: unknown;
}

const newLang = "pt";
const kingdom = "plants";

const newSpecies: Species[] = JSON.parse(
  await Deno.readTextFile(`data/${kingdom}_${newLang}.json`)
);

const oldSpecies: Species[] = JSON.parse(
  await Deno.readTextFile(`Species Lists/${kingdom}.json`)
);

const nodes = oldSpecies.map((species) => species.nodeId);
const newNodes = newSpecies.map((species) => species.nodeId);

// Check for new nodes not in old species
for (const newNode of newNodes) {
  if (!nodes.includes(newNode)) {
    const species = newSpecies[newNodes.indexOf(newNode)];
    console.log(`Node ${newNode} not found in old species`);
    console.log(species);
  }
}

const output: Species[] = [];

for (const species of oldSpecies) {
  const node = species.nodeId;
  if (newNodes.includes(node)) {
    const newAnimal = newSpecies[newNodes.indexOf(node)];
    try {
      const field = `name_${newLang}`;
      const newName = newAnimal[field];
      species[field] = newName;
    } catch (error) {
      console.log(`Field name_${newLang} not found in new species`);
      console.log(newAnimal);
    }
    output.push(species);
  } else {
    throw new Error(`Node ${node} not found in new ${kingdom}`);
  }
}

await Deno.writeTextFile(
  `data/${kingdom}_added.json`,
  JSON.stringify(output, null, 4)
);

console.log("Done");
